routes/api.js

```diff
@@ -1,29 +1,117 @@
+		// body sanity check
+		res.set({
+				'Content-Type': 'text/plain',
+				'Station-State': 'Disabled'
+		});
+		res.sendStatus(403);
+	};
+
+	if(!req.headers.badge || !req.headers.sid || !req.headers['station-state']) {
+		// check the necessary headers are provided for future processing
+		res.sendStatus(403);
+	};
+
+	// TODO check if user is already logged in to another machine
+	// the piece can be extracted because if the user is on another
+	// machine then there is no need to perform all the checks
+	// this will *might* require a module that contains all users
+	// currently loogged into a machine.
+	
+	var headerObj = req.headers;
+
+	db.getStation(headerObj.sid).then(function(station) {
+		//check if machine is in database
+		!station ? res.sendStatus(403) : console.log(station);
+		db.validateUser(headerObj.badge).then(function(user) {
+			//check if user accesing machine exist
+		 	!user ? res.sendStatus(403) : console.log(user);
+			if (user.dataValues.status === 'Admin') {
+				!user.dataValues.status ? res.sendStatus(503) : console.log(user.dataValues.status)
+				// check cache to see if there is a user on the machine
+				var userWaiting = cache.getMachine(headerObj.sid, timer);
+				if (userWaiting === null && headerObj['station-state'] === 'disabled') {
+					//there is no user in cache and the machine is 'disabled'
+					//put the admin on the machine	
+					// call station module HERE //to update station with current user and change to enabled
+					res.set({
+						'Content-Type': 'text/plain',
+						'Station-State': 'enabled',
+						'badge':  headerObj.badge
+					});
+					res.sendStatus(200);
+				} else if (headerObj['station-state'] === 'enabled') {
+					//assumption is the admin is logging of from the machine
+					res.set({
+						'Content-Type': 'text/plain',
+						'Station-State': 'disabled'
+					});
+					res.sendStatus(200);
+				} else {
+					// grant privilages to the user	
+					db.grantPrivileges(userWaiting, headerObj.sid).then(function(result) {
+						cache.delete(headerObj.sid);
+						//call station module HERE //to update station with current user and change to enabled
+						res.set("Connection", "close");
+						!result ? res.sendStatus(403) : res.set({
+																							'Content-Type': 'text/plain',
+																							'Station-state': 'enabled', 
+																							'badge': userWaiting
+																						});
+						// remove user from module
+						res.sendStatus(200);
+					});
+				};
+			} else if (user.dataValues.status === 'User' || user.dataValues.status === 'Manager') {
+				// handle user
+				// Assumption is the user is logging off the machine
+				//this needs to check if a user is on the machine
+				if (headerObj['station-state'] === 'enabled') {
+					// call station module HERE //to update station disabled and remove user
+					res.set({
+						'Content-Type': 'text/plain',
+						'Station-State': 'disabled'
+					});
+					res.sendStatus(200);
```

Comment:
I believe this allows any non-Admin user to log someone out of a station. Should be checking if the badge swipe is from the user who's currently logged in. If not, it's a 403 and no station-state change.